test/e2e/framework/pv/pv.go

```diff
@@ -800,3 +804,52 @@ func SkipIfNoDefaultStorageClass(c clientset.Interface) {
 		e2eskipper.Skipf("error finding default storageClass : %v", err)
 	}
 }
+
+// GetWindowsStorageClassName returns Windows storageClass or return error
+func GetWindowsStorageClassName(c clientset.Interface) (string, error) {
+	list, err := c.StorageV1().StorageClasses().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("Error listing storage classes: %v", err)
+	}
+	var scName string
+	for _, sc := range list.Items {
+		if IsWindowsAnnotation(sc.ObjectMeta) {
+			if len(scName) != 0 {
+				return "", fmt.Errorf("Multiple Windows storage classes found: %q and %q", scName, sc.Name)
+			}
+			scName = sc.Name
+		}
+	}
+	if len(scName) == 0 {
+		return "", fmt.Errorf("No Windows storage class found")
+	}
+	return scName, nil
+}
+
+// SkipIfNoWindowsStorageClass skips tests if no Windows SC can be found.
+func SkipIfNoWindowsStorageClass(c clientset.Interface) {
+	_, err := GetWindowsStorageClassName(c)
+	if err != nil {
+		e2eskipper.Skipf("error finding windows storageClass : %v", err)
+	}
+}
+
+// SkipIfNoStorageClass skips tests if no default SC for linux or Windows SC
+func SkipIfNoStorageClass(c clientset.Interface) {
+	if framework.NodeOSDistroIs("windows") {
```

Comment:
noting that while not new, this concept seems broken. e.g. what about mixed clusters ...?

Reply:
Even though the cluster could a mixed of OSes, NodeOSDistro is a string value set in TestContext. So it could be either "linux" or "windows" or others.. I think.